writers/postgres/messages.go

```diff
@@ -0,0 +1,116 @@
+//
+// Copyright (c) 2018
+// Mainflux
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package postgres
+
+import (
+	"github.com/jmoiron/sqlx"
+	"github.com/lib/pq" // required for DB access
+	"github.com/mainflux/mainflux"
+	"github.com/mainflux/mainflux/writers"
+)
+
+const (
+	errDuplicate = "unique_violation"
+	errFK        = "foreign_key_violation"
+	errInvalid   = "invalid_text_representation"
+)
+
+var _ writers.MessageRepository = (*postgresRepo)(nil)
+
+type postgresRepo struct {
+	db *sqlx.DB
+}
+
+// New returns new PostgreSQL writer.
+func New(db *sqlx.DB) writers.MessageRepository {
+	return &postgresRepo{db: db}
+}
+
+func (pr postgresRepo) Save(msg mainflux.Message) error {
+	q := `INSERT INTO messages (channel, subtopic, publisher, protocol,
+    name, unit, value, string_value, bool_value, data_value, value_sum,
+    time, update_time, link)
+    VALUES (:channel, :subtopic, :publisher, :protocol, :name, :unit,
+    :value, :string_value, :bool_value, :data_value, :value_sum,
+    :time, :update_time, :link);`
+
+	dbth := toDBMessage(msg)
+
+	_, err := pr.db.NamedExec(q, dbth)
+	if err != nil {
+		pqErr, ok := err.(*pq.Error)
+		if ok {
+			switch pqErr.Code.Name() {
+			case errInvalid:
+				return nil
+			}
```

Comment:
Not sure if this is the right way to handle this error.

Reply:
That's the way to check for PostgreSQL errors. I'm not sure about ignoring the "invalid_text_representation" errors. The protobuf and the go compiler will protect us from setting wrong string representations for the parameter values, but I think one extra check does not hurt.
@manuio What's the idea with ignoring this kind of errors?

Reply:
Yes, that is what I meant, `return nil` is just ignoring it. Also, can there be any other errors from Postgres?

Reply:
sure there can be other errors and I also think we should also "repack" the [pq.error](https://github.com/lib/pq/blob/master/error.go) error code and error message and return it as go error.

Reply:
Yes, I agree. @manuio you can check out `things` service implementation for example.

Reply:
Good catch, I removed the error and then forgot... I'll fix it

Reply:
Good catch, I removed and then forgot... I'll fix it